File: C++/DataStructures/bfs.cpp
#include <iostream>
#include<map>
#include<bits/stdc++.h>
using namespace std;
template<typename T>
class Graph
{
    map<T,list<T>>l;
    public:
    void addedge(int x,int y)
    {
        l[x].push_back(y);
        l[y].push_back(x);
    }
    void bfs(T src)
    {
        map<T,int>visited;
        queue<T>q ;
        q.push(src);
        visited[src]=true;
        while(!q.empty())
        {
            T node=q.front();
            q.pop();
            cout<<node<<" ";
            for(int nbr:l[node])
            {
                if(!visited[nbr])
                {
                    q.push(nbr);
                    visited[nbr]=true;
                }
            }
        }
    }
};
        
int main() {
	// your code goes here
	Graph<char>ob;
	ob.addedge('A','B');
	ob.addedge('A','D');
	
	ob.addedge('A','C');
	ob.addedge('C','D');
	ob.addedge('D','E');
	ob.addedge('E','F');
	ob.bfs('A');
	return 0;
}

File: Python/Algorithms/Graphs/bfs.py
from collections import defaultdict
class Graph:
    def __init__(self):
        self.graph=defaultdict(list)
    def addEdge(self,u,v):
        self.graph[u].append(v)
    def printg(self):
        print(self.graph)
    def bfs(self,s):
        q=[]
        visited=[False]*len(self.graph)
        q.append(s)
        visited[s]=True
        while q:
            k=q.pop(0)
            print(k,end=" ")
            for i in self.graph[k]:
                if visited[i]==False:
                    q.append(i)
                    visited[i]=True
g=Graph()
g.addEdge(0, 1) 
g.addEdge(0, 2) 
g.addEdge(1, 2) 
g.addEdge(2, 0) 
g.addEdge(2, 3) 
g.addEdge(3, 3)
g.printg()
g.bfs(2)
